node/overseer/src/lib.rs

```diff
@@ -1035,6 +1040,57 @@ mod tests {
 		}
 	}
 
+	// item isn't equal, but is close enough
+	//
+	// we need this to properly test that we've collected all expected results, even if
+	// they arrive out of order
+	trait Equivalent {
+		fn equiv(&self, other: &Self) -> bool;
+	}
+
+	impl Equivalent for ActiveLeavesUpdate {
+		fn equiv(&self, other: &Self) -> bool {
+			self.activated.iter().collect::<HashSet<_>>() == other.activated.iter().collect::<HashSet<_>>() &&
+			self.deactivated.iter().collect::<HashSet<_>>() == other.deactivated.iter().collect::<HashSet<_>>()
+		}
+	}
+
+	impl Equivalent for OverseerSignal {
+		fn equiv(&self, other: &Self) -> bool {
+			use OverseerSignal::{
+				ActiveLeaves,
+				Conclude,
+			};
+
+			match (self, other) {
+				(ActiveLeaves(lalu), ActiveLeaves(ralu)) => lalu.equiv(ralu),
+				(Conclude, Conclude) => true,
+				_ => false,
+			}
+		}
```

Comment:
although these are send around as `Vec`s essentially they are sets so it would probably be cleaner to just implement `Eq` on `ActiveLeavesUpdate`?